### PR TITLE
Update method

### PR DESCRIPTION
**Pull request by misazr**

### CODE REVIEW

File: lib/rest_in_peace.rb [hunk 1]
```diff
@@ -47,10 +47,8 @@ def clear_changes
   end
 
   def update_attributes(attributes)
-    attributes.each do |key, value|
-      next unless respond_to?("#{key}=")
-      send("#{key}=", value)
-    end
+    warn 'DEPRECATION WARNING: the method update_attributes is deprecated, use method update.'
```
Comment (calmyournerves) [inline lib/rest_in_peace.rb]: Now that you moved it to `active_model_api`, why deprecate this? We certainly need it for classes that should not be ActiveModel compatible.

File: spec/rest_in_peace/definition_proxy/resource_method_definitions_spec.rb [hunk 2]
```diff
@@ -127,14 +127,14 @@
   context '#put' do
     it_behaves_like 'an instance method' do
       let(:http_verb) { :put }
-      let(:method_name) { :update }
+      let(:method_name) { :update_a }
```
Comment (calmyournerves) [inline spec/rest_in_peace/definition_proxy/resource_method_definitions_spec.rb]: Why this change?

Comment (misazr) [inline spec/rest_in_peace/definition_proxy/resource_method_definitions_spec.rb]: Because it was calling method `update` which is new one, here should calls just link to the path to put method.

Comment (calmyournerves) [inline spec/rest_in_peace/definition_proxy/resource_method_definitions_spec.rb]: Ah, maybe you can give it a meaningful name? Someone else looking at the code might be confused.

File: lib/rest_in_peace.rb (hunk 1 above)
Comment (misazr) [inline lib/rest_in_peace.rb]: https://redmine.nine.ch/issues/32737#Acceptance-Criteria last point, David was required to have `deprecated message`

File: spec/rest_in_peace/definition_proxy/resource_method_definitions_spec.rb (hunk 2 above)
Comment (misazr) [inline spec/rest_in_peace/definition_proxy/resource_method_definitions_spec.rb]: Ok, will rename with `update_record`

File: lib/rest_in_peace.rb (hunk 1 above)
Comment (calmyournerves) [inline lib/rest_in_peace.rb]: Yeah, but that was under the assumption that we can replace it. We can't, that's why we moved it to `active_model_api.rb`. The existing behaviour should be left as is.

Comment (misazr) [inline lib/rest_in_peace.rb]: ok, I understand.
